builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -373,12 +374,16 @@ func (p *puller) resolve(ctx context.Context, g session.Group) error {
 			if err != nil {
 				return struct{}{}, err
 			}
-			newRef, _, dt, err := p.is.ResolveImageConfig(ctx, ref.String(), llb.ResolveImageConfigOpt{Platform: &p.platform, ResolveMode: p.src.ResolveMode.String()}, p.sm, g)
+			_, dt, err := p.is.ResolveImageConfig(ctx, ref.String(), sourceresolver.Opt{
+				Platform: &p.platform,
+				ImageOpt: &sourceresolver.ResolveImageOpt{
+					ResolveMode: p.src.ResolveMode.String(),
+				},
+			}, p.sm, g)
 			if err != nil {
 				return struct{}{}, err
 			}
 
-			p.ref = newRef
```

Comment:
Is this correct?

Reply:
Oh, good catch!

Yeah it looks like this shouldn't be deleted.
I restored it as:
`p.ref = ref.String()`

because `ResolveImageConfig` no longer returns a reference.
Looking at the old implementation, it didn't really change the ref that was passed as an argument.
The only place where it returned a different ref is where it went to the `(remotes.Resolver) Resolve()`, but it doesn't seem to change it either (at least the `docker` remote).